src/rockstor/fs/btrfs.py

```diff
@@ -934,24 +934,57 @@ def scrub_start(pool, force=False):
 
 
 def scrub_status(pool):
+    """
+    Returns the raw statistics per-device (-R option) of the ongoing or last
+    known btrfs scrub. Works by parsing the output of the following command:
+    btrfs scrub status -R <mount-point>
+    :param pool: pool object
+    :return: dictionary indexed via 'status' and if a finished or halted, or
+    cancelld scrub is indicated then the duration of that scrub is added as
+    value to added index 'duration'. In all 'status' cases bar 'unknown',
+    data_bytes_scrubbed is passed as value to index 'kb_scrubbed' and all
+    other -R invoked details are returned as key value pairs.
+    """
     stats = {'status': 'unknown', }
     mnt_pt = mount_root(pool)
     out, err, rc = run_command([BTRFS, 'scrub', 'status', '-R', mnt_pt])
-    if (len(out) > 1):
-        if (re.search('running', out[1]) is not None):
+    if err != [''] and len(err) > 0:
```

Comment:
Thanks, I'll revisit this soon as I have a few tidy/checkup's/todo's pending now and have in mind a suitable issue against which I can do them. But I think it's right as your analogy would hold if it were err != [] as len([]) = 0, however len(['']) = 1 and I think I was after:

if not an empty string in list (specifically) and at least a single element (to ensure a safe 0 index).

But yes given we then check the exact contents of err[0] directly there after the first if clause is redundant. However given it's the most common return (err=[''] is the norm) it's a rapid and cheap first check to avoid all others: I had imagined we would have occasion to add additional inner ifs than the one I added there.

I can at least clarify with comments on the why.

Cheers.